### PR TITLE
Queue RPCs by Worker_EntityId

### PR DESCRIPTION
**Pull request by aleximprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
RPCs are now guaranteed to arrive in the same order per target actor, i.e. for the actor and all of its subobjects.

#### Release note
RPCs are now guaranteed to arrive in the same order per target actor, i.e. for the actor and all of its subobjects.

#### Tests
RPC ordering should be the same as in original UE4 implementation.

#### Documentation
RPCs are now guaranteed to arrive in the same order per target actor, i.e. for the actor and all of its subobjects.

#### Primary reviewers
@improbable-valentyn @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h [hunk 1]
```diff
@@ -27,11 +27,11 @@ class FRPCContainer
 public:
 	void QueueRPC(FPendingRPCParamsPtr Params, ESchemaComponentType Type);
 	void ProcessRPCs(const FProcessRPCDelegate& FunctionToApply);
-	bool ObjectHasRPCsQueuedOfType(const FUnrealObjectRef& TargetObjectRef, ESchemaComponentType Type) const;
+	bool ObjectHasRPCsQueuedOfType(const Worker_EntityId& EntityId, ESchemaComponentType Type) const;
 
 private:
 	using FArrayOfParams = TArray<FPendingRPCParamsPtr>;
-	using FRPCMap = TMap<FUnrealObjectRef, FArrayOfParams>;
+	using FRPCMap = TMap<Worker_EntityId, FArrayOfParams>;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h]:
```suggestion
	using FRPCMap = TMap<Worker_EntityId_Key, FArrayOfParams>;
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp [hunk 2]
```diff
@@ -54,12 +54,12 @@ void FRPCContainer::ProcessRPCs(const FProcessRPCDelegate& FunctionToApply)
 	}
 }
 
-bool FRPCContainer::ObjectHasRPCsQueuedOfType(const FUnrealObjectRef& TargetObjectRef, ESchemaComponentType Type) const
+bool FRPCContainer::ObjectHasRPCsQueuedOfType(const Worker_EntityId& EntityId, ESchemaComponentType Type) const
 {
 	const FRPCMap* MapOfQueues = QueuedRPCs.Find(Type);
 	if(MapOfQueues)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]:
```suggestion
	if (MapOfQueues)
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp [hunk 3]
```diff
@@ -54,12 +54,12 @@ void FRPCContainer::ProcessRPCs(const FProcessRPCDelegate& FunctionToApply)
...
 	{
-		const FArrayOfParams* RPCList = MapOfQueues->Find(TargetObjectRef);
+		const FArrayOfParams* RPCList = MapOfQueues->Find(EntityId);
 		if(RPCList)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]:
```suggestion
		if (RPCList)
```

File: SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h [hunk 4]
```diff
@@ -27,11 +27,11 @@ class FRPCContainer
 public:
 	void QueueRPC(FPendingRPCParamsPtr Params, ESchemaComponentType Type);
 	void ProcessRPCs(const FProcessRPCDelegate& FunctionToApply);
-	bool ObjectHasRPCsQueuedOfType(const FUnrealObjectRef& TargetObjectRef, ESchemaComponentType Type) const;
+	bool ObjectHasRPCsQueuedOfType(const Worker_EntityId& EntityId, ESchemaComponentType Type) const;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h]:
```suggestion
	bool ObjectHasRPCsQueuedOfType(Worker_EntityId EntityId, ESchemaComponentType Type) const;
```
A matter of preference

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp [hunk 5]
```diff
@@ -54,13 +54,13 @@ void FRPCContainer::ProcessRPCs(const FProcessRPCDelegate& FunctionToApply)
 	}
 }
 
-bool FRPCContainer::ObjectHasRPCsQueuedOfType(const FUnrealObjectRef& TargetObjectRef, ESchemaComponentType Type) const
+bool FRPCContainer::ObjectHasRPCsQueuedOfType(const Worker_EntityId& EntityId, ESchemaComponentType Type) const
 {
 	const FRPCMap* MapOfQueues = QueuedRPCs.Find(Type);
-	if(MapOfQueues)
+	if (MapOfQueues)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]: Driveby: Can you move the declaration inside the `if` statement.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp [hunk 6]
```diff
@@ -54,13 +54,13 @@ void FRPCContainer::ProcessRPCs(const FProcessRPCDelegate& FunctionToApply)
...
 	{
-		const FArrayOfParams* RPCList = MapOfQueues->Find(TargetObjectRef);
-		if(RPCList)
+		const FArrayOfParams* RPCList = MapOfQueues->Find(EntityId);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]: Likewise here

File: SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h (hunk 1 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h]: Changed.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp (hunk 5 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]: Updated.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp (hunk 6 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp]: Updated.
